core/src/test/java/org/apache/iceberg/avro/TestSchemaConversions.java

```diff
@@ -76,7 +78,8 @@ public void testPrimitiveTypes() {
             Schema.createFixed("fixed_12", null, null, 12),
             Schema.create(Schema.Type.BYTES),
             LogicalTypes.decimal(9, 4)
-                .addToSchema(Schema.createFixed("decimal_9_4", null, null, 4)));
+                .addToSchema(Schema.createFixed("decimal_9_4", null, null, 4)),
+            variant("rnull"));
```

Comment:
Why `rnull`?

Reply:
Ah, I see. That's because there was no field ID in conversion. I think this is alright, but I would prefer a separate test case that converts the field inside a record so that we get a reasonable name. I don't think this is incorrect name behavior, but it isn't something that we need to verify with an assertion.

I would also prefer to catch the case where the ID is null and use a generic name, like `variant`.